src/main/java/io/confluent/kafkarest/v2/KafkaConsumerManager.java

```diff
@@ -0,0 +1,440 @@
+      }
+    }
+
+    // Ensure we clean up the placeholder if there are any issues creating the consumer instance
+    boolean succeeded = false;
+    try {
+      log.debug("Creating consumer " + name + " in group " + group);
+
+      // Note the ordering here. We want to allow overrides, but almost all the
+      // consumer-specific settings don't make sense to override globally (e.g. group ID, consumer
+      // ID), and others we want to ensure get overridden (e.g. consumer.timeout.ms, which we
+      // intentionally name differently in our own configs).
+      Properties props = (Properties) config.getOriginalProperties().clone();
+      //props.setProperty("zookeeper.connect", zookeeperConnect);
+      props.setProperty(KafkaRestConfig.BOOTSTRAP_SERVERS_CONFIG, this.bootstrapServers);      
+      props.setProperty("group.id", group);
+      // This ID we pass here has to be unique, only pass a value along if the deprecated ID field
+      // was passed in. This generally shouldn't be used, but is maintained for compatibility.
+      if (instanceConfig.getId() != null) {
+        props.setProperty("consumer.id", instanceConfig.getId());
+      }
+      // To support the old consumer interface with broken peek()/missing poll(timeout)
+      // functionality, we always use a timeout. This can't perfectly guarantee a total request
+      // timeout, but can get as close as this timeout's value
+      props.setProperty("consumer.timeout.ms", ((Integer) iteratorTimeoutMs).toString());
+      if (instanceConfig.getAutoCommitEnable() != null) {
+        props.setProperty("auto.commit.enable", instanceConfig.getAutoCommitEnable());
+      } else {
+        props.setProperty("auto.commit.enable", "false");
+      }
+      if (instanceConfig.getAutoOffsetReset() != null) {
+        props.setProperty("auto.offset.reset", instanceConfig.getAutoOffsetReset());
+      }
+
+      props.setProperty("schema.registry.url",
+			config.getString(KafkaRestConfig.SCHEMA_REGISTRY_URL_CONFIG));
+      
+      switch (instanceConfig.getFormat()) {
+      case AVRO:
+	  props.put("key.deserializer", "io.confluent.kafka.serializers.KafkaAvroDeserializer");
+	  props.put("value.deserializer", "io.confluent.kafka.serializers.KafkaAvroDeserializer");
+	  break;
+      case JSON:
+      case BINARY:
+      default:
+	  props.put("key.deserializer", "org.apache.kafka.common.serialization.ByteArrayDeserializer");
+	  props.put("value.deserializer", "org.apache.kafka.common.serialization.ByteArrayDeserializer");
+      }
+
+      Consumer consumer = null;
+      try {
+        if (consumerFactory == null) {
+	    consumer = new KafkaConsumer(props);
+        } else {
+	    consumer = consumerFactory.createConsumer(props);
+        }
+      } finally {
+      }
+
+      KafkaConsumerState state = null;
+      switch (instanceConfig.getFormat()) {
+        case BINARY:
+          state = new BinaryKafkaConsumerState(this.config, cid, consumer);
```

Comment:
Related to the above comment, I notice that these are all duplicated and share a different abstract base class for v2 compared to v1. Might be worth thinking about whether a separate class will be necessary (or easier), or if it's worth trying to share some of the implementation (since I'd imagine at least a few of the core operations will be shared, e.g. lifecycle stuff like expiration, polling for data, committing offsets, etc.. (I'm not really sure about this, but in some other tools like the command line tools for Kafka we have tried to unify the two APIs behind another layer of abstraction and it seems to have worked ok, see e.g. the console consumer for an example of this.)